src/components/Dropdown/Dropdown.tsx

```diff
@@ -162,20 +192,41 @@ function useFocusContextInstance(
   lastTypedOrSubmittedValue: string,
   setValue: (newValue: string) => void,
   screenReaderKey: number,
-  setScreenReaderKey: (newKey: number) => void
+  setScreenReaderKey: (newKey: number) => void,
+  alwaysSelectOption: boolean,
 ): FocusContextType {
   const [focusedIndex, setFocusedIndex] = useState(-1);
   const [focusedValue, setFocusedValue] = useState<string | null>(null);
   const [focusedItemData, setFocusedItemData] = useState<Record<string, unknown> | undefined>(undefined);
+  useEffect(() => {
+    if (alwaysSelectOption) {
+      if (items.length > 0) {
+        const index = focusedIndex === -1 || focusedIndex >= items.length ? 0 : focusedIndex;
+        setFocusedIndex(index);
+        setFocusedValue(items[index].value);
```

Comment:
can we refactor all the setBlah into one helper function and invoke that instead?